pkg/webhook/controlplane/ensurer.go

```diff
@@ -40,13 +44,33 @@ func (e *ensurer) InjectClient(client client.Client) error {
 
 // EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
 func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, ectx genericmutator.EnsurerContext, new, _ *appsv1.Deployment) error {
+	cluster, _ := ectx.GetCluster(ctx)
+	makeAuditForwarder := false
+
+	if e.controllerConfig.ClusterAudit.Enabled {
+		cpConfig, err := helper.ControlPlaneConfigFromClusterShootSpec(cluster)
+		if err != nil {
+			logger.Error(err, "Could not read ControlPlaneConfig from cluster shoot spec", "Cluster name", cluster.ObjectMeta.Name)
+		}
+		if cpConfig.FeatureGates.ClusterAudit != nil && *cpConfig.FeatureGates.ClusterAudit {
+			makeAuditForwarder = true
+		}
+	}
+
 	template := &new.Spec.Template
 	ps := &template.Spec
 	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
-		ensureKubeAPIServerCommandLineArgs(c, e.controllerConfig)
-		ensureVolumeMounts(c, e.controllerConfig)
-		ensureVolumes(ps, e.controllerConfig)
+		ensureKubeAPIServerCommandLineArgs(c, makeAuditForwarder, e.controllerConfig)
+		ensureVolumeMounts(c, makeAuditForwarder, e.controllerConfig)
+		ensureVolumes(ps, makeAuditForwarder, e.controllerConfig)
+	}
+	if makeAuditForwarder {
+		err := ensureAuditForwarder(ps, e.controllerConfig)
+		if err != nil {
+			logger.Error(err, "Could not ensure the audit forwarder", "Cluster name", cluster.ObjectMeta.Name)
```

Comment:
```suggestion
			logger.Error(err, "Could not ensure the audit forwarder", "Cluster name", cluster.ObjectMeta.Name)
			return err
```

Reply:
See above - I am not sure what happens if we do not return the expected result; and I would not want to create more breakage because one feature fails.